packages/expo-device/ios/DeviceModule.swift

```diff
@@ -0,0 +1,98 @@
+// swiftlint:disable:all number_separator
+
+import ExpoModulesCore
+import UIKit
+import MachO
+
+public class DeviceModule: Module {
+  public func definition() -> ModuleDefinition {
+    Name("ExpoDevice")
+
+    Constants([
+      "isDevice": isDevice(),
+      "brand": "Apple",
+      "manufacturer": "Apple",
+      "modelId": UIDevice.modelIdentifier,
+      "modelName": UIDevice.DeviceMap.modelName,
+      "deviceYearClass": UIDevice.DeviceMap.deviceYearClass,
+      "totalMemory": ProcessInfo.processInfo.physicalMemory,
+      "osName": UIDevice.current.systemName,
+      "osVersion": UIDevice.current.systemVersion,
+      "osBuildId": osBuildId(),
+      "osInternalBuildId": osBuildId(),
+      "deviceName": UIDevice.current.name,
+      "supportedCpuArchitectures": cpuArchitectures()
+    ])
+
+    AsyncFunction("getDeviceTypeAsync") { () -> Int in
+      switch UIDevice.current.userInterfaceIdiom {
+      case UIUserInterfaceIdiom.phone:
+        return DeviceType.phone.rawValue
+      case UIUserInterfaceIdiom.pad:
+        return DeviceType.tablet.rawValue
+      case UIUserInterfaceIdiom.tv:
+        return DeviceType.tv.rawValue
+      default:
+        return DeviceType.unknown.rawValue
+      }
+    }
+
+    AsyncFunction("getUptimeAsync") { () -> Double in
+      let uptimeMs: Double = ProcessInfo.processInfo.systemUptime * 1000
+
+      return uptimeMs
+    }
+
+    AsyncFunction("isRootedExperimentalAsync") { () -> Bool in
+      return UIDevice.current.isJailBroken
+    }
+  }
+}
+
+func isDevice() -> Bool {
+  #if targetEnvironment(simulator)
+    return false
+  #else
+    return true
+  #endif
+}
+
+func osBuildId() -> String? {
+  #if os(tvOS)
+    return nil
+  #else
+    // Credit: https://stackoverflow.com/a/65858410
+    var mib: [Int32] = [CTL_KERN, KERN_OSVERSION]
+    let namelen = u_int(MemoryLayout.size(ofValue: mib) / MemoryLayout.size(ofValue: mib[0]))
```

Comment:
Let's use types from Swift instead of C 😉

```suggestion
    let namelen = UInt32(MemoryLayout.size(ofValue: mib) / MemoryLayout.size(ofValue: mib[0]))
```